task_sdk/tests/execution_time/test_supervisor.py

```diff
@@ -862,6 +862,14 @@ def watched_subprocess(self, mocker):
                 "",
                 id="patch_task_instance_to_skipped",
             ),
+            pytest.param(
+                TaskState(state=TerminalTIState.FAILED, end_date=timezone.parse("2024-10-31T12:00:00Z")),
+                b"",
+                "",
+                (),
+                "",
+                id="patch_task_instance_to_failed",
+            ),
```

Comment:
Don't think we need this since it is already covered by `TaskState` via -- since all we need to check is it can handle `TaskState` msg:

```
TaskState(state=TerminalTIState.SKIPPED, end_date=timezone.parse("2024-10-31T12:00:00Z")),
```

Reply:
Yea that's true but I kept this change to be a little explicit about which state it is going through explicitly. Would it be nice to just create a loop inside params about this instead or nuke it entirely?

Reply:
I think nuking it is fine imo